In [2]:
%pip install openpyxl
import pandas as pd
# Load the dataset
df = pd.read_excel("tp_2017conference.xlsx")

# Preview
df.head()

Note: you may need to restart the kernel to use updated packages.


title  \
0  Making Neural Programming Architectures Genera...   
1  Making Neural Programming Architectures Genera...   
2  Making Neural Programming Architectures Genera...   
3  End-to-end Optimized Image Compression | OpenR...   
4  End-to-end Optimized Image Compression | OpenR...   

                            Unnamed: 1  \
0  06 Nov 2016 (modified: 11 Mar 2017)   
1  06 Nov 2016 (modified: 11 Mar 2017)   
2  06 Nov 2016 (modified: 11 Mar 2017)   
3  06 Nov 2016 (modified: 03 Mar 2017)   
4  06 Nov 2016 (modified: 03 Mar 2017)   

                                            abstract  \
0  Abstract:###Empirically, neural networks that ...   
1  Abstract:###Empirically, neural networks that ...   
2  Abstract:###Empirically, neural networks that ...   
3  Abstract:###We describe an image compression m...   
4  Abstract:###We describe an image compression m...   

                    keywords                                            E   F  \
0  Keywords:###Deep learning                    Conflicts:###berkeley.edu  77   
1  Keywords:###Deep learning                    Conflicts:###berkeley.edu  77   
2  Keywords:###Deep learning                    Conflicts:###berkeley.edu  77   
3  Keywords:###Deep learning  Conflicts:###nyu.edu, rwth-aachen.de, uv.es  51   
4  Keywords:###Deep learning  Conflicts:###nyu.edu, rwth-aachen.de, uv.es  51   

                                     G                   decision  \
0                          02 May 2017  Decision:###Accept (Oral)   
1                          02 May 2017  Decision:###Accept (Oral)   
2                          02 May 2017  Decision:###Accept (Oral)   
3  20 Feb 2017 (modified: 20 Feb 2017)  Decision:###Accept (Oral)   
4  20 Feb 2017 (modified: 20 Feb 2017)  Decision:###Accept (Oral)   

                                                   I    J  \
0  Comment:###The reviewers were very favourable,...  129   
1  Comment:###The reviewers were very favourable,...  129   
2  Comment:###The reviewers were very favourable,...  129   
3  Comment:###This is one of the two top papers i...  169   
4  Comment:###This is one of the two top papers i...  169   

                                                   K  \
0                            17 Dec 2016 23 Dec 2016   
1                            17 Dec 2016 23 Dec 2016   
2  16 Dec 2016 16 Dec 2016 23 Dec 2016 (modified:...   
3                20 Feb 2017 (modified: 20 Feb 2017)   
4                03 Jan 2017 (modified: 03 Jan 2017)   

                                          confidence  \
0  Confidence:###4: The reviewer is confident but...   
1  Confidence:###3: The reviewer is fairly confid...   
2  Confidence:###5: The reviewer is absolutely ce...   
3  Confidence:###4: The reviewer is confident but...   
4  Confidence:###4: The reviewer is confident but...   

                                                rate   N   O   P   Q   R  \
0  Rating:###8: Top 50% of accepted papers, clear... NaN NaN NaN NaN NaN   
1  Rating:###8: Top 50% of accepted papers, clear... NaN NaN NaN NaN NaN   
2  Rating:###9: Top 15% of accepted papers, stron... NaN NaN NaN NaN NaN   
3                    Rating:###7: Good paper, accept NaN NaN NaN NaN NaN   
4  Rating:###9: Top 15% of accepted papers, stron... NaN NaN NaN NaN NaN   

                                              review     T  
0  This paper argues that being able to handle re...  1665  
1  This is a very interesting and fairly easy to ...  1024  
2  This paper improves significantly upon the ori...   566  
3  Two things I*d like to see. 1) Specifics about...   698  
4  This is the most convincing paper on image com...   761

In [8]:
# Compute statistics
num_papers = df["title"].nunique()
num_reviews = int(df["review"].count())

# Average tokens per paper (based on title + abstract)
df["paper_text"] = df["title"].astype(str) + " " + df["abstract"].astype(str)
avg_tokens_per_paper = df["paper_text"].apply(lambda x: len(x.split())).mean()

# Average tokens per review
avg_tokens_per_review = df["review"].apply(lambda x: len(x.split())).mean()

# Show computed statistics
stats = {
    "Number of papers": num_papers,
    "Number of reviews": num_reviews,
    "Avg. tokens per paper": round(avg_tokens_per_paper),
    "Avg. tokens per review": round(avg_tokens_per_review),
}
stats

{'Number of papers': 489,
 'Number of reviews': 1495,
 'Avg. tokens per paper': 163,
 'Avg. tokens per review': 296}

The dataset used for fine-tuning consists of 489 unique papers and 1,495 peer reviews. On average, each paper contains 163 tokens (title + abstract), while reviews contain about 296 tokens.